x/ibc/handler.go

```diff
@@ -21,7 +20,7 @@ func NewHandler(k Keeper) sdk.Handler {
 			return client.HandleMsgCreateClient(ctx, k.ClientKeeper, msg)
 
 		case client.MsgUpdateClient:
-			return client.HandleMsgUpdateClient(ctx, k.ClientKeeper, msg)
+			return sdk.Result{} // XXX: add events either here or in ante
```

Comment:
IMO events should happen wherever the state updates do (so then in ante)

Reply:
why did we change the update client logic? that's unrelated to ADR 15

Reply:
`UpdateClient` execution is moved to the antehandler, so we don't need to handle them inside the app handler

Reply:
Is this the right path forward here?

Reply:
Yes `MsgUpdateClient` has the route `ibc` and this handler will handle it